server/controllers/pet.js

```diff
@@ -39,34 +39,42 @@ exports.createPet = asyncHandler(async (req, res, next) => {
       newPet.petPhoto = petPhoto._id;
       await newPet.save();
     }
-    if (req.files) {
-      let images = req.files;
-
-      images.forEach(async (image) => {
-        const result = await cloudinary.uploader.upload(image.path);
-
-        const newImage = await Image.create({
-          imageUrl: result.secure_url,
-          cloudinaryId: result.public_id,
-        });
-        newPet.photoGallery.push(newImage._id);
-      });
-      await newPet.save();
-    }
+    let updatedPet = await Pet.findById(newPet._id).populate('petPhoto');
     res.status(201).json({
       success: {
-        pet: newPet,
+        pet: updatedPet,
       },
     });
   }
 });
-
+exports.addPetPhotoGallery = asyncHandler(async (req, res, next) => {
+  let { petId } = req.body;
+  let currentPet = await Pet.findById(petId);
+  let images = req.files;
```

Comment:
It's good to first validate the request body so that you can return a `400` if any required field is missing.